### ISSUE TITLE
Contribute this back to Xonotic or Flathub?

### ISSUE THREAD

**Issue by nsrosenqvist** [casept/flatpak-xonotic#1]
Hi, just wondering if you have any plans on getting this integrated with flathub so that it's made available for others easily?

**Comment by nsrosenqvist** [comment]
@casept have you looked into this?

**Comment by skylar-eng** [comment]
Agree. Someone needs to send this upstream, and/or see if we can get the ball rolling to put it on flathub.

**Comment by casept** [comment]
I'm not really familiar with the xonotic developers/part of the dev community, so I don't really know who/how to approach about this. Also, the xonotic project advertises "no-installation" play by downloading and extracting the .zip from their site, so offering additional official packages might be seen as confusing for new users. 

There are also technical reasons why I don't think the package is fit for upstreaming, the primary one being the fact that it doesn't do much other than repackage the upstream release zipball. This means that it's not possible to package nightly releases, for example. The reason for not building xonotic from source is the project's obtuse build system which I couldn't get to work with this package even after spending an entire weekend trying.

I also don't know how much work it would be to integrate the package with the upstream build system/CI and other infrastructure, and so on. I don't really feel like putting the work into it myself, and just dropping the work onto upstream developers seems a bit rude to me. Flathub also strives to only accept packages maintained by the upstream developers, which means I can't get it onto flathub either.

That said, if someone wants to fix the package and prepare it for upstreaming be my guest. The files are under the unlicense, so I think pushing them upstream shouldn't cause any issues on the licensing front no matter who does it.

**Comment by nsrosenqvist** [comment]
I created an issue on the Xonotic bug tracker to see what the official developers have to say about it: https://gitlab.com/xonotic/xonotic/issues/214
